### ISSUE TITLE
Completion of constructors should insert parentheses

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#37563]
Try completing after "Dir" in the following code:
```dart
import 'dart:io';

void f() {
  Dir
}
```

The list includes "Directory" and "Directory()", the first presumably being the class name and the second presumably being the unnamed constructor. Selecting either one inserts just the class name (expect for the first, but not the second). Similarly, the list includes "Directory.fromRawPath()", but selecting it inserts everything other than the parentheses. This seems wrong.

If you choose a completion corresponding to a method you get the parentheses inserted automatically, so it seems consistent, and helpful, to include the parentheses after a constructor invocation. Also, it would match what's being shown in the list of suggestions.

**Comment by bwilkerson** [comment]
@scheglov @lambdabaa

**Comment by lambdabaa** [comment]
Based on [what the client does](https://github.com/JetBrains/intellij-plugins/blob/master/Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java#L449) I suspect we need a change to `completion.getSuggestionDetails`.

**Comment by scheglov** [comment]
I don't think so.
The `completion.getSuggestionDetails` is only for updating imports.
The `AvailableSuggestion` already includes enough information about constructor to insert it correctly.

```json
          {
            "label": "Directory",
            "declaringLibraryUri": "dart:io",
            "element": {
              "kind": "CONSTRUCTOR",
              "name": "",
              "location": {
                "file": "/Users/scheglov/Applications/dart-sdk/lib/io/directory.dart",
                "offset": 4860,
                "length": 0,
                "startLine": 131,
                "startColumn": 11
              },
              "flags": 0,
              "parameters": "(String path)",
              "returnType": "Directory"
            },
            "defaultArgumentListString": "path",
            "defaultArgumentListTextRanges": [
              0,
              4
            ],
            "docComplete": "Creates a [Directory] object.\n\nIf [path] is a relative path, it will be interpreted relative to the\ncurrent working directory (see [Directory.current]), when used.\n\nIf [path] is an absolute path, it will be immune to changes to the\ncurrent working directory.",
            "docSummary": "Creates a [Directory] object.",
            "parameterNames": [
              "path"
            ],
            "parameterTypes": [
              "String"
            ],
            "requiredParameterCount": 1
          },
```

**Comment by lambdabaa** [comment]
The relevant change just landed in https://github.com/JetBrains/intellij-plugins/commit/f0ed86542d98ac7f5bdd85f1bc2ddd8e0f87b4bf

**Comment by lambdabaa** [comment]
This issue is fixed in Dart plugin version [191.8348](https://plugins.jetbrains.com/plugin/6351-dart/update/66572).
